### ISSUE TITLE
Warning regarding proj@7 when installing GDAL and Other Failures

### ISSUE THREAD

**Issue by OldSubSailor** [Homebrew/homebrew-core#82441]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
No Logs
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I am installing GDAL as part of the process for installing Ortho4XPlane.

### What happened (include all command output)?

GDAL installs, but I get a warning about proj@7 during the installation process. 
Warning: proj@7 has been deprecated because it is not supported upstream!

After instillation, I see another concern listed when I brew info
% brew info GDAL
gdal: stable 3.3.1 (bottled), HEAD
Geospatial Data Abstraction Library
https://www.gdal.org/
Conflicts with:
  avce00 (because both install a cpl_conv.h header)
  cpl (because both install cpl_error.h)

also when I try to call GDAL from python

>>> import GDAL
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'GDAL'
>>> 

GDAL is shown on brew listing
% brew list
==> Formulae
cairo			gdal	

### What did you expect to happen?

I expected no warnings. 
I expected the gdal module to be found. 
Because GDAL module is not found even when called in Python, it does not work in the Ortho4Xplane application. 
Hence, GDAL is a constant fail all around. 

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install GDAL
brew list
brew info GDAL
```

**Comment by cho-m** [comment]
The warning on `proj@7` is expected and can be ignored. As long as `gdal` and all its dependencies don't support `proj@8`, there is not much Homebrew can do but wait for upstream support.

Conflicts with again are not failures but mainly informational due to projects having similar files/binaries/etc. Shouldn't cause an issue unless you need to install 2 formulae that conflict.

For Python bindings, make sure you are using Homebrew Python and use the latest API, e.g.
```console
❯ /usr/local/bin/python3 -c 'from osgeo import gdal; print(gdal.__version__)'
3.3.1
```

---

Based on my own test, everything is working as expected and there isn't a bug here. The warnings are informational details and the Python issue seems to be that you are trying to use the older API.

**Comment by OldSubSailor** [comment]
Thank you very much for replying. I hope that I made it clear I know next to nothing about this topic. I am indeed old as my user name indicates. I have been asking for help in various places in Github and have been told to submit pull request to various places. When I have been able to submit a pull request, I usually get a rather terse, unhelpful reply, when I actually need more of a left foot, right foot helping hand.
With regard the Python bindings, and I am not a programmer, they came from what I could best determine, a line of Python code from (https://github.com/oscarpilote/Ortho4XP) when an error was generated while running the Ortho4XP application.

So from what I understand that you have told me so far is that I need to NOT use Python from python.org, but rather download it from Homebrew. Now if I recall correctly, Python was installed when I installed Homebrew GDAL, also it was installed via Xcode and Command Line. I seem to have all sorts of flavors of Python lingering about. Again, if I remember correctly in an effort to get Ortho4XP to run under Big Sur and all the Gatekeeper security issues. 

So am I on the right path at this point, and do I need to start out by deleting all the Pythons except the default 2.7 and Xcode and Command Line tools? Perhaps cleaning house with regard to Homebrew also?

And 
In looking at /usr/local/bin there is no python3 listed but there are a lot of gdal files listed, for example:
gdal-config -> ../Cellar/gdal/3.3.1_2/bin/gdal-config
gdal2tiles.py -> ../Cellar/gdal/3.3.1_2/bin/gdal2tiles.py
gdal2xyz.py -> ../Cellar/gdal/3.3.1_2/bin/gdal2xyz.py
gdal_calc.py -> ../Cellar/gdal/3.3.1_2/bin/gdal_calc.py
gdal_contour -> ../Cellar/gdal/3.3.1_2/bin/gdal_contour
gdal_create -> ../Cellar/gdal/3.3.1_2/bin/gdal_create
gdal_edit.py -> ../Cellar/gdal/3.3.1_2/bin/gdal_edit.py
gdal_fillnodata.py -> ../Cellar/gdal/3.3.1_2/bin/gdal_fillnodata.py
gdal_grid -> ../Cellar/gdal/3.3.1_2/bin/gdal_grid

**Comment by OldSubSailor** [comment]
Thank you very much for replying. I hope that I made it clear I know next to nothing about this topic. I am indeed old. I have been asking for help in various places in Github and have been told to submit pull request to various places. When I have been able to submit a pull request, I usually get a rather terse, unhelpful reply, when I actually need more of a left foot, right foot helping hand.
With regard the Python bindings, and I am not a programmer, they came from what I could best determine, a line of Python code from (https://github.com/oscarpilote/Ortho4XP) when an error was generated while running the Ortho4XP

**Comment by cho-m** [comment]
Another thing you can try is manually editing the Python file, though it is not guaranteed to work.

1. Go to directory you put Ortho4XP
2. Open the file `src/O4_DEM_Utils.py`
3. Change the line `import gdal` to `from osgeo import gdal`.

Not sure if other places need to be modified.

**Comment by OldSubSailor** [comment]
Again thank you very much. I appreciate all you have done to help me.
At this point Ortho4XP does work except for the USGS 1/3" & 1"TILE creations which requires GDAL. So I guess that we will have to live with that until someone decides that they want to become the new Ortho Wizard. 

I am confused about the api stuff. I went to the link that you provided and saw what you said about the import not working any more. But  using brew install GDAL does not install that osgeo "flavor". 
I am getting this
>>> from osgeo import gdal
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'osgeo'

**Comment by OldSubSailor** [comment]
a locate search does show this for example
/usr/local/lib/python3.9/site-packages/osgeo_utils/samples/gdal_lut.py

**Comment by cho-m** [comment]
Have you tried checking what python3 is being used?

For example, if you set the PATH to use Apple's Python3, it will fail:
```console
$ export PATH="/usr/bin:/bin:/usr/sbin:/sbin"

$ which python3
/usr/bin/python3

$ python3 -c 'from osgeo import gdal'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'osgeo'
```

If you add Homebrew's Python as first (which should automatically be done by using `brew shellenv`), it should work:
```console
$ eval $(/usr/local/bin/brew shellenv)

$ which python3
/usr/local/bin/python3

$ python3 -c 'from osgeo import gdal'

$ python3 -c 'from osgeo import gdal; print(gdal.__version__)'
3.3.1
```

**Comment by OldSubSailor** [comment]
I have deleted all the various pythons that were on my iMac except 2.7. Then uninstalled Homebrew, Xcode and command line tools. I reinstalled Homebrew, which reinstalled command line tools and Python 3.8 from Apple. I then installed Homebrew python (3.9.6), and the Homebrew files required by Ortho4XP except GDAL. However, for some reason I also had to install Homebrew tk3.9. to get Ortho4XP to work. This is something that I do not have to do if I use Python from Python.org. 

% which python3
/usr/local/bin/python3
which I suppose is this:  python3 -> ../Cellar/python@3.9/3.9.6/bin/python3
from a long listing of the directory

% python3
Python 3.9.6 (default, Jun 29 2021, 05:25:02) 

Do I need to be changing any path stuff now? 
% echo $PATH
/Users/opt/miniconda3/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/opt/local/bin:/opt/local/sbin:/Library/Frameworks/Python.framework/Versions/3.9/bin:/Library/Frameworks/Python.framework/Versions/3.8/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin

I think that miniconda is a leftover trial.


So I do believe that I am using Homebrew python at this time. The only other version of Python3 is 3.8 in the command line tools, and I have not seen it show up. 

So if I use brew install GDAL, I should get the osgeo version that should work with the import?

**Comment by OldSubSailor** [comment]
Well you did it!! Thank you. I took the last step and installed GDAL, and from the terminal in python 3.9 ran... from osgeo import gdal and got no error message.
Then ran:
% python3 -c 'from osgeo import gdal; print(gdal.__version__)'
3.3.1
3.3.1
zsh: command not found: 3.3.1
% 
Which I took as good to go. So I edited the Ortho4XP file as you suggested and ran Ortho4XP and created a tile using USGS 1/3" which calls GDAL.  **SUCCESS**
I be going back to the X-plane Ortho for X-plane forum and writing new instructions for Big Sur.  I would like to give you credit if you do not mind. 
This even if folks do not want to go through the extra steps of GDAL inclusion, just the simplification of the process via Homebrew only will be a big help to the Big Sur crowd, alas, Gatekeeper security will remain a major pain.
